# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1292306564 28800
#      Mon Dec 13 22:02:44 2010 -0800
# Node ID e4ba3cb2d27ef67b4ea5fa3fa3900d9bb8dc4ea9
# Parent  f5ee3e6e926603e78fedda780277c85ebb6b28bb
isfield: Additional test for cellstring of field names

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-12-13  Rik  <octave@nomad.inbox5.com>
+
+	* ov-struct.cc (isfield): Additional test for cellstring of field names.
+
 2010-12-13  Rik  <octave@nomad.inbox5.com>
 
 	* data.cc (mod): Additional tests for mod with non-integer real values.
 
 2010-12-12  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/lookup.cc (Flookup): Drop support for the "i" option.
 
diff --git a/src/ov-struct.cc b/src/ov-struct.cc
--- a/src/ov-struct.cc
+++ b/src/ov-struct.cc
@@ -2019,22 +2019,22 @@ Return the number of fields of the struc
 
   return retval;
 }
 
 /*
 %!# test isfield
 %!test
 %!  x(3).d=1; x(2).a=2; x(1).b=3; x(2).c=3;
-%!  assert (isfield (x, "b"));
-%!assert( isfield( struct("a", "1"), "a"));
-%!assert( isfield( {1}, "c"), logical (0));
-%!assert( isfield( struct("a", "1"), 10), logical (0));
-%!assert( isfield( struct("a", "b"), "a "), logical (0));
-
+%!  assert (isfield (x, 'b'));
+%!assert (isfield (struct('a', '1'), 'a'));
+%!assert (isfield ({1}, 'c'), false);
+%!assert (isfield (struct('a', '1'), 10), false);
+%!assert (isfield (struct('a', 'b'), "a "), false);
+%!assert (isfield (struct('a', 1, 'b', 2), {'a', 'c'}), [true, false]);
 */
 
 DEFUN (cell2struct, args, ,
        "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} cell2struct (@var{cell}, @var{fields}, @var{dim})\n\
 Convert @var{cell} to a structure.  The number of fields in @var{fields}\n\
 must match the number of elements in @var{cell} along dimension @var{dim},\n\
 that is @code{numel (@var{fields}) == size (@var{cell}, @var{dim})}.\n\
